weldr: accept rhsm parameter in sources

msehnout opened the pull request:
The system sources allow specification of the rhsm parameter, but it
isn't available in the sources configured over the Weldr API. This patch
implements support for it.

Fixes: #1445


This pull request includes:

- [ ] adequate testing for the new functionality or fixed issue
- [x] adequate documentation informing people about the change such as
  - [x] create a file in [news/unreleased](https://github.com/osbuild/osbuild-composer/tree/main/docs/news/unreleased) directory if this change should be mentioned in the release news
  - [ ] submit a PR for the [guides](https://github.com/osbuild/guides) repository if this PR changed any behavior described there: https://www.osbuild.org/guides/

<!--
Thanks for proposing a change to osbuild-composer!

Please don't remove the above check list. These are things that each pull
request must have before it is merged. It helps maintainers to not forget
anything.

If the reason for ticking any of the boxes is ambiguous, please add a short
note explaining why.

For user-visible changes, "adequate documentation" is an entry describing the
change for users in docs/news. Please refer to docs/news/README.md for details.

In addition, if this pull request fixes a downstream issue, please refer to
test/README.md and add these additional items:

- [ ] 1st commit of any `rhbz#` related PR contains bug reproducer; CI reports FAIL or
- [ ] PR contains automated tests for new functionality and
- [ ] QE has approved reproducer/new tests and
- [ ] Subsequent commits provide bug fixes without modifying the reproducer; CI reports PASS and
- [ ] QE approves this PR; RHBZ status is set to `MODIFIED + Verified=Tested`
-->

teg reviewed (approved):
This makes sense to me! Solves a real problem of adding non standard repos from the RH CDN. Would be great if Brian could give it a sanity check too though.

bcl commented on internal/weldr/json.go:
```diff
@@ -139,6 +139,7 @@ type SourceConfigV0 struct {
 	System   bool     `json:"system" toml:"system"`
 	Proxy    string   `json:"proxy,omitempty" toml:"proxy,omitempty"`
 	GPGUrls  []string `json:"gpgkey_urls,omitempty" toml:"gpgkey_urls,omitempty"`
+	RHSM     bool     `json:"rhsm" toml:"rhsm"`
```
Technically this should only be added to the V1 Source structs/json since V0 is 'finished'.

msehnout replied:
Where is V0 used?

bcl replied:
/api/v0/projects/source/* uses it.

msehnout replied:
Sure, but how does one use it with composer-cli? I tried it on RHEL and it uses the V1 version.

Anyhow, I've made the changes you requested. Is it ok now?

bcl replied:
composer-cli uses the v1 route, but the v0 routes and responses should be left alone since they are 'done'.

bcl reviewed (changes requested):
GH won't let you comment on other bits of code, so I'll put this here:

RHSM value also needs to be copied in other functions that convert to/from the store's structure and the JSON API structure. I *think* the complete list is:

* SourceConfigV1.SourceConfig()
* NewSourceConfigV1

bcl reviewed (dismissed):
Looks good to me.

msehnout commented:
Added a test case and a news entry.

teg commented:
This is looking great to me! @atodorov do you have any thoughts on the testing of this? There is a nice unit test I think, but not sure what integration test (if any) it makes sense to hook into to verify this? The RT or Ansible repositories are candidates we could test against, but I think we don't currently test against the CDN in our CI, so this may not make sense?

atodorov reviewed (changes requested):
I don't see any changes related to the test suite/new test scenarios being added. At the very minimum just to make sure assignments of values continue working across the different functions. 

At the same time I see slightly decreased test coverage, for example in internal/store/json.go an if condition that was covered before is now not covered. Given there are no other changes in the PR that may be a result of the new field in the struct. Please check it out.

msehnout commented:
> I don't see any changes related to the test suite/new test scenarios being added. At the very minimum just to make sure assignments of values continue working across the different functions.
> 
> At the same time I see slightly decreased test coverage, for example in internal/store/json.go an if condition that was covered before is now not covered. Given there are no other changes in the PR that may be a result of the new field in the struct. Please check it out.

You are right, I thought this PR already contained a test case. Maybe I misremember.

Anyway, I've introduced one additional unit test and added a verification to the weldr client tests. Is it ok like this?
